celavi/pylca_celavi/des_interface.py

```diff
@@ -188,72 +193,77 @@ def pylca_run_main(self, df):
-                        lca_db = lca_db[['year','stage','material','flow name','emission factor kg/kg']]
-                        lca_db = lca_db[lca_db['material'] != 'concrete']
-                        lca_db['year'] = lca_db['year'].astype(int)
-                        lca_db = lca_db.drop_duplicates()
-                        lca_db.to_csv(self.shortcutlca_filename,
-                                      mode = 'a',
-                                      index = False,
-                                      header = False)
+                        # model_celavi_lci() is calculating foreground processes and dynamics of electricity mix.
+                        # It calculates the LCI flows of the foreground process.
+                        res = model_celavi_lci(working_df,year,facility_id,stage,material,state,df_static,self.dynamic_lci_filename,self.electricity_grid_spatial_level,self.intermediate_demand_filename)
+                        # model_celavi_lci_insitu() calculating direct emissions from foreground
+                        # processes.
+                        emission = model_celavi_lci_insitu(working_df,year,facility_id,stage,material,df_emissions)
         
-            else:
-                print(str(facility_id) + ' - ' + str(year) + ' - ' + stage + ' - ' + material + ' shortcut calculations done',flush = True)    
-
-            res_df = pd.concat([res_df,result_shortcut])
+                        if not res.empty:
+                            res = model_celavi_lci_background(res,year,facility_id,stage,material,self.uslci_filename,self.lci_activity_locations)
+                            lci = postprocessing(res,emission)
+                            res = impact_calculations(lci,self.traci_lci_filename)
+                            res_df = pd.concat([res_df,res])
+                            lcia_mass_flow = pd.concat([lci,lcia_mass_flow])
+                            
+                            
+                            df_with_no_lca_entry = df_with_no_lca_entry.drop(['flow name'],axis = 1)
+                            lca_db = df_with_no_lca_entry.merge(lcia_mass_flow,on = ['year','stage','material'])
+                            lca_db['emission factor kg/kg'] = lca_db['flow quantity_y']/lca_db['flow quantity_x']   
+                            lca_db = lca_db[['year','stage','material','flow name','emission factor kg/kg']]
+                            lca_db = lca_db[lca_db['material'] != 'concrete']
+                            lca_db['year'] = lca_db['year'].astype(int)
+                            lca_db = lca_db.drop_duplicates()
+                            lca_db.to_csv(self.shortcutlca_filename,
+                                          mode = 'a',
+                                          index = False,
+                                          header = False)
+                        else:        
+                           print(f'LCIA optimization failed for {material} at {stage} in {year}')
```

Comment:
I updated this print statement so it's more informative and can be used for debugging. However now, with either the national or state grid mix, I'm getting output messages like:

```
Final demand for 2033 next use glass fiber is zero
LCIA optimization failed for glass fiber at next use in 2033
```

So I don't think the if/else statement structures here are correct, because it's saying the optimization failed when no optimization should have run anyway.

Reply:
I actually made some changes to the code to have print statements for explicit cases of 

1. No listing in the LCA inventory, thus returning empty dataframe. 
2. No final demand, flows are zero. 
3. Actual failure of optimization due to some error. 
The case for next use and in use of materials falls in category 1. Thus for that category, we get two print statements - 
a. There is not listing of LCA inventory for this material and stage and b, the LCIA code returned an empy dataframe. 

These changes have been incorporated in PR for issue #113 

> The "LCIA optimization failed" print statement (line 261 in des_interface) is being shown when final demand for a material in a stage is zero, meaning the calculations shouldn't be run anyway. That statement should either be moved or maybe deleted - there are many other warnings for failed optimizations and it's not clear if this one is necessary.
> 
> When I'm running without shortcut calculations, I am getting a few other failed optimizations - see my other comment. Landfilling fails sporadically when using the state level grid mix and manufacturing fails consistently when using the national grid mix. Since this is the first time I'm running with the failure warnings, I don't know if these are to be expected or not. I recommend testing and looking into the LCI results to see if there's actually a problem.
> 
> I'm approving, so you can merge in once you've moved or deleted the LCIA optimization failed print statement referenced above. If there is any additional coding needed to deal with the landfilling and manufacturing LCI failures, the I would like to re-review before merging in.

1. When the final demand is zero, I have added and explicit statement that just says final demand is zero and nothing happens. For cases like in use, next use, final demand is not zero. There is a final demand but due to no listing in the LCA Inventory, no calculations occur. I have added an explicit print statement highlighting this situation. 
2.